test-framework/emulator_backend.go

```diff
@@ -0,0 +1,159 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2022 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Based on https://github.com/wk8/go-ordered-map, Copyright Jean Rougé
+ *
+ */
+
+package test
+
+import (
+	emulator "github.com/onflow/flow-emulator"
+
+	"github.com/onflow/cadence/encoding/json"
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/errors"
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/onflow/cadence/runtime/stdlib"
+	"github.com/onflow/cadence/runtime/tests/utils"
+)
+
+var _ interpreter.TestFramework = &EmulatorBackend{}
+
+// EmulatorBackend is the emulator-backed implementation of the interpreter.TestFramework.
+//
+type EmulatorBackend struct {
+	blockchain *emulator.Blockchain
+}
+
+func NewEmulatorBackend() *EmulatorBackend {
+	return &EmulatorBackend{
+		blockchain: newBlockchain(),
+	}
+}
+
+func (e *EmulatorBackend) RunScript(code string, args []interpreter.Value) interpreter.ScriptResult {
+	// TODO: maybe re-use interpreter? Only needed for value conversion
```

Comment:
Value conversion uses the interpreter for its Elaboration. Is there a potential conflict from reusing it here? Like if a contract is updated between calls.

I know we do some kind of cache invalidation but I'm not sure if it applies here.